src/api.v2/controllers/sampleFileController.js

```diff
@@ -25,19 +25,42 @@ const createFile = async (req, res) => {
     upload_status: 'uploading',
   };
 
-  let uploadUrlParams;
+
 
   await sqlClient.get().transaction(async (trx) => {
     await new SampleFile(trx).create(newSampleFile);
     await new Sample(trx).setNewFile(sampleId, sampleFileId, sampleFileType);
-
-    logger.log(`Getting multipart upload urls for ${experimentId}, sample ${sampleId}, sampleFileType ${sampleFileType}`);
-    uploadUrlParams = await getFileUploadUrls(sampleFileId, metadata, size, bucketNames.SAMPLE_FILES);
   });
 
 
   logger.log(`Finished creating sample file for experiment ${experimentId}, sample ${sampleId}, sampleFileType ${sampleFileType}`);
-  res.json(uploadUrlParams);
+
+  res.json(OK());
+};
+
+const beginUpload = async (req, res) => {
+  const {
+    params: { experimentId, sampleFileId },
+    body: { metadata, size },
+  } = req;
+
+  const { uploadStatus } = await new SampleFile().findById(sampleFileId).first();
+  // Check that the file is already in the process of being uploaded
+  // If it isn't, then it can't be reuploaded, a new file should be created
+  // This is because the existing files may be shared across many experiments
+  if (!['uploading', 'compressing'].includes(uploadStatus)) {
+    throw new MethodNotAllowedError(
+      `Sample file ${sampleFileId} is not in the process of being uploaded. 
+      No sample files can be replaced in s3, to replace a file referenced in a sample, create a new file for it`,
+    );
+  }
```

Comment:
The original motivation behind making the createFile endpoint return the signed urls was to make it harder for us to overwrite existing sample files which are shared across many different experiments. 

In order to maintain this security, I added this check. Only files that are in the process of being uploaded (`'uploading'` or `'compressing'`) can have multipart uploads initiated for them

Reply:
can you explain a bit more why it's either `uploading` or `compressing`? I'm not familiar with the different status.

Reply:
If you look at the possible statuses in the db you'll see these options:

'CREATE TYPE upload_status AS ENUM (\'invalidFormat\', \'uploaded\', \'uploading\', \'compressing\', \'uploadError\', \'fileNotFound\', \'fileReadError\', \'fileReadAborted\');',

Of these, only 2 of them imply that an upload is ongoing. That's why I chose only those 2

Reply:
I don't understand why compressing is considered upload, isn't that happening in the UI?

Reply:
it's considered part of the upload process, we could remove it if that's what you were thinking about (although not in this PR I think)